cubed/runtime/executors/dask.py

```diff
@@ -0,0 +1,48 @@
+import dask
+
+from cubed.core.array import TaskEndEvent
+from cubed.core.plan import visit_nodes
+from cubed.runtime.types import DagExecutor
+
+
+def exec_stage_func(func, *args, **kwargs):
+    # TODO would be good to give the dask tasks useful names
+    return dask.delayed(func(*args, **kwargs))  # should we add pure=True?
```

Comment:
Looks like you are invoking the function when `exec_stage_func` is called?
```suggestion
    return dask.delayed(func)(*args, **kwargs)  # should we add pure=True?
```

Reply:
OOPS

Reply:
I'm still seeing spilling

![Screenshot from 2023-06-28 16-14-12](https://github.com/tomwhite/cubed/assets/35968931/ec193ed0-6233-45eb-88c8-0321b4423e35)